aicsimageio/readers/bioformats_reader.py

```diff
@@ -0,0 +1,457 @@
+
+        if series is not None:
+            self._r.setSeries(series)
+
+        nt, nc, nz, ny, nx, nrgb = self.core_meta.shape
+        chunks = ((1,) * nt, (1,) * nc, (1,) * nz, ny, nx)
+        if nrgb > 1:
+            chunks = chunks + (nrgb,)
+        arr = da.map_blocks(
+            self._dask_chunk,
+            chunks=chunks,
+            dtype=self.core_meta.dtype,
+        )
+        return _DaskArrayProxy(arr, self)
+
+    @property
+    def is_open(self) -> bool:
+        """Whether the underlying file is currently open"""
+        return bool(self._r.getCurrentFile())
+
+    @property
+    def filename(self) -> str:
+        """Return name of file handle."""
+        # return self._r.getCurrentFile()
+        return self._path
+
+    @property
+    def ome_xml(self) -> str:
+        """return OME XML string."""
+        with self:
+            store = self._r.getMetadataStore()
+
+            return str(store.dumpXML()) if store else ""
+
+    @property
+    def ome_metadata(self) -> "OME":
+        """Return OME object parsed by ome_types."""
+        import ome_types
+
+        from ..metadata import utils as metadata_utils
+
+        xml = metadata_utils.clean_ome_xml_for_known_issues(self.ome_xml)
+        return ome_types.from_xml(xml)
+
+    def __enter__(self) -> "LociFile":
+        if not self.is_open:
+            self.open()
+        return self
+
+    def __exit__(self, *args: Any) -> None:
+        self.close()
+
+    def __del__(self) -> None:
+        self.close()
+
+    def _get_plane(
+        self, t=0, c=0, z=0, y: slice = slice(None), x: slice = slice(None)
+    ) -> np.ndarray:
+        """Load a single plane."""
+        idx = self._r.getIndex(z, c, t)
+        *_, ny, nx, nrgb = self.core_meta.shape
+        ystart, ywidth = _slice2width(y, ny)
+        xstart, xwidth = _slice2width(x, nx)
+        with self._lock:
+            buffer = self._r.openBytes(idx, xstart, ystart, xwidth, ywidth)
+            # TODO: check what the bare minimim amount of copying is to prevent
+            # reading incorrect regions (doesnt' segfault, but can get weird)
+            im = np.frombuffer(buffer[:], self.core_meta.dtype).copy()
+        if nrgb > 1:
+            # TODO: check this with some examples
+            if self.core_meta.is_interleaved:
+                im.shape = (ywidth, xwidth, nrgb)
+            else:
+                im.shape = (nrgb, ywidth, xwidth)
+                im = np.transpose(im, (1, 2, 0))
+        else:
+            im.shape = (ywidth, xwidth)
+        # TODO: check this... we might need to reshaped the non interleaved case
+        return im
+
+    def _dask_chunk(self, block_id: tuple) -> np.ndarray:
+        # if someone indexes a 5D dask array as `arr[0,1,2]`, then the `info`
+        # dict will contain the key: 'chunk-location': (0, 1, 2, 0, 0)
+        # reader.getIndex() expects (Z, C, T) ...
+        # We ASSUME that the final dask array is in the order TCZYX, so chunk-location
```

Comment:
for now, that's an internal assumption (and it just makes things easier).  as far as our interaction with bioformats is concerned, it all really comes down to `openBytes` call [here](https://github.com/AllenCellModeling/aicsimageio/pull/306/files#diff-2d684fcc182480aea8c37c6dc033094a287076307dd0796601d42d1a7e281259R319) where you just give it an ZCT index and it gets it for you.  As we discussed, I'm honestly not quite sure what the relevance of `getDimensionOrder` is.  that is: it doesn't impose limitations on how you retrieve the planes.  So for now I've just been saying "stick with TCZYX", ignore native dimensionOrder (until we learn why we might not want to), and let the end-user reshape our xarray if they want to

Reply:
I think the problem with that is that our most common interaction is `get_image_data` / `get_image_dask_data` which takes in a dimension string instead of integer / slice indices.

if the backing data has the data mapped to dimensions incorrectly then I would assume the `get_image_data` would return incorrectly mapped data too.

Reply:
Maybe I am misunderstanding though. Are you storing the data in our native order or in the files native order? If you are storing the data in our native order then the above concern is invalid.

Reply:
let me look into it a bit.  my intention was to provide the data always in the native order... as a 5d  TCZYX array if not RGB or a 6D `TCZYXc` if rgb.  I basically ignore the file's native order.

Reply:
I think it's totally reasonable that a Bioformats reader would give back TCZYX because it is itself a layer on top of proprietary file format readers.  Just as long as the metadata is consistent with that...

Reply:
> my intention was to provide the data always in the native order... as a 5d TCZYX array if not RGB or a 6D TCZYXc if rgb. I basically ignore the file's native order.

that is totally fine with me 👍